### PR TITLE
Usecases

### PR DESCRIPTION
**Pull request by Cotto4**
At least shell code for the test cases, some/all full implementation

### CODE REVIEW

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/RemoteDataStore.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	public void pushQuestion(Question mQuestion) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	public void pushAnswer(Answer mAnswer) {
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/RemoteDataStore.java]: Don't like the word "Push". Doesn't make sense for local which should have the same interface. "Put" instead.

File: QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/DataManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package ca.ualberta.cs.cmput301f14t14.questionapp.test;
+
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.Answer;
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.DataManager;
```
Comment (stephenjust) [inline QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/DataManagerTest.java]: DataManager probably shouldn't be in the model package...

File: QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/RemoteDataStoreTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package ca.ualberta.cs.cmput301f14t14.questionapp.test;
+
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.Answer;
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.Comment;
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.DataManager;
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.Image;
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.Question;
+import ca.ualberta.cs.cmput301f14t14.questionapp.model.RemoteDataStore;
```
Comment (stephenjust) [inline QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/RemoteDataStoreTest.java]: Also should not be in model.

**Comment by Cotto4** [comment]
Got it, fixed the put names. Where should they go, their own package maybe?

**Comment by stephenjust** [comment]
Root package is fine. Maybe "data" or something.

## 

Stephen Just

**Comment by Cotto4** [comment]
Move datastores too? Or just the manager?

**Comment by stephenjust** [comment]
Both

**Comment by Cotto4** [comment]
Nailed it.

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Question.java [hunk 4]
```diff
@@ -1,11 +1,21 @@
...
 public class Question extends Model {
-
-	public Question(String title, String body, Image image) {
+	
+	private Integer mId;
+	private String mTitle;
+	private String mBody;
+	private Image mImage;
+	
+	public Question(String t, String b, Image i) {
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/model/Question.java]: Why did you change this line? It makes eclipse's autocomplete especially useless. (I assume it was because the var names conflicted with class attributes.)

**Comment by stephenjust** [comment]
Approved. @linster: Review?

**Comment by linster** [comment]
Looks good. The tests already fail, so there's no way I can break the build :)
